samples/mpsl/timeslot/src/main.c

```diff
@@ -120,12 +132,11 @@ static mpsl_timeslot_signal_return_param_t *mpsl_timeslot_callback(
 		break;
 	}
 
-	/* Put callback info in the message queue. */
-	int err = k_msgq_put(&callback_msgq, &signal_type, K_NO_WAIT);
-
-	if (err) {
-		error();
-	}
+#if defined(NRF5340_XXAA_NETWORK)
+	NVIC_SetPendingIRQ(SWI2_IRQn);
+#elif defined(NRF52840_XXAA) || defined(NRF52832_XXAA) || defined(NRF52832_XXAB)
+	NVIC_SetPendingIRQ(SWI2_EGU2_IRQn);
+#endif
```

Comment:
If you use `CONFIG_SOC_SERIES_NRF52X` or `CONFIG_SOC_SERIES_NRF53X`, the code becomes more future proof. Also, the build should fail if the board is unknown. 

```suggestion
#if defined(NRF5340_XXAA_NETWORK)
	NVIC_SetPendingIRQ(SWI1_IRQn);
#elif defined(CONFIG_SOC_SERIES_NRF52X)
	NVIC_SetPendingIRQ(SWI1_EGU1_IRQn);
#else
#error Unknown device
#endif
```
I would suggest to use `SWI1_EGU1_IRQn` on nRF52 series devices as `SWI2_EGU2_IRQn` does not exist on certain devices in this series.